### PR TITLE
Add Edit gist api

### LINKED ISSUE #238: Implement Edit a gist API
Edit a gist API is not implemented:
https://developer.github.com/v3/gists/#edit-a-gist

### PR DESCRIPTION
**Pull request by aleksandr-vin**
Closes #238

### CODE REVIEW

File: github4s/shared/src/main/scala/github4s/api/Gists.scala [hunk 1]
```diff
@@ -56,4 +58,50 @@ class Gists[C, M[_]](
...
+      gistId: String,
+      description: String,
+      files: Map[String, Option[EditGistFile]],
+      headers: Map[String, String] = Map(),
+      accessToken: Option[String] = None): M[GHResponse[Gist]] = {
+
+    implicit val encodeEditGistFile: Encoder[EditGistFile] = {
+      deriveEncoder[EditGistFile].mapJsonObject(_.filter(e =>
+        !(e._1.equals("filename") && e._2.isNull)))
+    }
```
Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/Gists.scala]: You can move that to `Encoders.scala` :+1:

Comment (aleksandr-vin) [reply github4s/shared/src/main/scala/github4s/api/Gists.scala]: I tried first, but it seems like _Encoders_ is not used here, when `toJson` is needed for implicit.

Comment (BenFradet) [reply github4s/shared/src/main/scala/github4s/api/Gists.scala]: I think you should be able to import it then, no?

Comment (aleksandr-vin) [reply github4s/shared/src/main/scala/github4s/api/Gists.scala]: Done.

**Comment by juanpedromoreno** [comment]
@aleksandr-vin Could you please rebase against `upstream` master? It's weird, but locally I'm getting this compilation issue:

```bash
error] /github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala:202: illegal start of simple expression
[error]     )
[error]     ^
[error] /github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala:215: ')' expected but '}' found.
[error]   }
[error]   ^
[error] two errors found
[error] (github4sJVM/test:compileIncremental) Compilation failed
```

**Comment by aleksandr-vin** [comment]
> @aleksandr-vin Could you please rebase against `upstream` master? It's weird, but locally I'm getting this compilation issue:
> 
> ```shell
> error] /github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala:202: illegal start of simple expression
> [error]     )
> [error]     ^
> [error] /github4s/shared/src/test/scala/github4s/unit/ReposSpec.scala:215: ')' expected but '}' found.
> [error]   }
> [error]   ^
> [error] two errors found
> [error] (github4sJVM/test:compileIncremental) Compilation failed
> ```
> 
> _Update_: it seems it happens for Scala 2.11.

@juanpedromoreno done.

**Comment by aleksandr-vin** [comment]
Closes #237 as well

**Comment by aleksandr-vin** [comment]
What's are your release plans, guys?

**Comment by BenFradet** [comment]
we usually do them on demand, so if you need one just shout :+1:

**Comment by juanpedromoreno** [comment]
@aleksandr-vin @BenFradet  I'll send a fix for the random test failure, and then, I'll request a new release along with the build upgrade.

**Comment by aleksandr-vin** [comment]
I need one, as I'm in the process of some home project and need an update on heroku (that's why #242) ;)

**Comment by juanpedromoreno** [comment]
Done https://github.com/47deg/github4s/pull/243

The release should be completed shortly ;)

**Pull request merged**